app/src/main/java/com/beemdevelopment/aegis/importers/AuthenticatorPlusImporter.java

```diff
@@ -0,0 +1,171 @@
+package com.beemdevelopment.aegis.importers;
+
+import android.content.Context;
+import android.database.Cursor;
+
+import com.beemdevelopment.aegis.R;
+import com.beemdevelopment.aegis.encoding.Base32;
+import com.beemdevelopment.aegis.encoding.EncodingException;
+import com.beemdevelopment.aegis.otp.HotpInfo;
+import com.beemdevelopment.aegis.otp.OtpInfo;
+import com.beemdevelopment.aegis.otp.OtpInfoException;
+import com.beemdevelopment.aegis.otp.TotpInfo;
+import com.beemdevelopment.aegis.ui.Dialogs;
+import com.beemdevelopment.aegis.vault.VaultEntry;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+public class AuthenticatorPlusImporter extends DatabaseImporter {
+    private static final String _pkgName = "com.mufri.authenticatorplus";
+
+    public AuthenticatorPlusImporter(Context context) {
+        super(context);
+    }
+
+    @Override
+    protected String getAppPkgName() {
+        return _pkgName;
+    }
+
+    @Override
+    protected String getAppSubPath() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public State read(FileReader reader) throws DatabaseImporterException {
+        return new EncryptedState(reader);
+    }
+
+    public static class EncryptedState extends DatabaseImporter.State {
+        private FileReader _reader;
+
+        private EncryptedState(FileReader reader) {
+            super(true);
+            _reader = reader;
+        }
+
+        @Override
+        public void decrypt(Context context, DecryptListener listener) {
+            // store InputStream in byte array
+            byte[] streamCopy = new byte[0];
```

Comment:
This is not great, but that's not your fault. SQL databases really throw a wrench in the DatabaseImporter abstraction we currently have.

Can you move the stream reading block inside ``State.read``, and pass the resulting byte array as a parameter to the EncryptedState constructor? You can also make the stream reading part shorter with the ``_reader.readAll()`` helper.

Reply:
This whole stream copying was just implemented, as the `SqlImportHelper.read` requires an `InputStream` as parameter. I therefore needed to pass an instance of either the `FileReader` or `InputStream` to this method, which is called in the `Dialogs.showPasswordInputDialog` lambda. As using some kind of stream inside a lambda seems to close it (I'm not that big of a Java guy, so don't know whats happening there), I passed a byte array and used it to recreate a stream inside the lambda.
I think it would be nicer, to modify the new `SqlImportHelper` classes to be able to handle both natively, so no trickery has to be done here.

Reply:
I understand why you wrote it that way. As you say, the stream is closed before that ``Dialogs.showPasswordInputDialog`` callback is called, so we have to read it into memory first.

Reply:
Would you then keep passing a steam, or modify the parameters as I suggested?

Reply:
If you can think of a nice way to do the latter, that's fine with me.